### ISSUE TITLE
Routing issues for Cars

### ISSUE THREAD

**Issue by karatemuffin** [organicmaps/organicmaps#9752]
I found a case where OrganicMaps Car Routing does something strange, while all 3 of the Routers from osm.org are generating expected results.  Also i checked OSM Data and was unable to find a reason for the behaviour. 

**Description**
I start from the [Schöcklhöhenstraße](https://www.openstreetmap.org/way/343512189) in Styria (Austria) just below the [Johann-Waller-Hütte](https://www.openstreetmap.org/way/225312785)  and want a route (car) to the hut.

**Expected behaviour**
I would expect it to use the following highways:

- Schöcklhöhenstraße Part 1: https://www.openstreetmap.org/way/343512189
- Schöcklhöhenstraße Part 2: https://www.openstreetmap.org/way/42686849
- Track to the Hut: https://www.openstreetmap.org/way/26728668

While i can see the expected results on all 3 routing algorithms on [osm.org](https://www.openstreetmap.org/directions?engine=fossgis_valhalla_car&route=47.186235%2C15.451645%3B47.189620%2C15.449455#map=17/47.187700/15.450178) the results on OrganicMaps are quite different. 

**Steps to reproduce**

The "routing_car" image shows that OM (as car) is routing me trought the woods then uses a highway and then again trought the woods to the hut. 
![routing_car](https://github.com/user-attachments/assets/034fb5c9-1a2d-46f8-9ecf-c5b896ed0eac)

If i drive a bicycle it chooses the path also to be expected for the car.
![routing_bycicle](https://github.com/user-attachments/assets/618fdd83-e299-4cf3-af5b-dfb697019a4d)

**System information:**
 - Operating system and its version: [Android 13]
 - Organic Maps version: [y2024.11.12-7-Google]
 - Device Model: [Unihertz Jelly Star]

**Comment by matkoniecz** [comment]
`motor_vehicle=private` is likely causing OM to ignore road and in desperation use what you see

is https://www.openstreetmap.org/way/343512195 by any chance missing `motor_vehicle=private`  ?

**Comment by matkoniecz** [comment]
Also, maybe in such case routing car on `motor_vehicle=private` should be allowed (at initial and final segment of route)

**Comment by karatemuffin** [comment]
> `motor_vehicle=private` is likely causing OM to ignore road and in desperation use what you see
> 
> is https://www.openstreetmap.org/way/343512195 by any chance missing `motor_vehicle=private` ?
Yes, those forestal highways should all be private, that i will fix in OSM, retry and report back. 

> Also, maybe in such case routing car on `motor_vehicle=private` should be allowed (at initial and final segment of route)
To keep it simple maybe just put a routing option like `allow private streets=off` equivalent to the `avoid motorways=off`?

**Comment by matkoniecz** [comment]
> Yes, those forestal highways should all be private, that i will fix in OSM, retry and report back.

note that you will need to wait for map data update

**Comment by matkoniecz** [comment]
> To keep it simple maybe just put a routing option like allow private streets=off equivalent to the avoid motorways=off?

I suspect that it should "just work" in general, and there is no real demand for taking shortcuts via private access roads in the middle of route

**Comment by karatemuffin** [comment]
> > Yes, those forestal highways should all be private, that i will fix in OSM, retry and report back.
> 
> note that you will need to wait for map data update
Yeah i got time :)

I opened a thread in the local [OSM Community Forum](https://community.openstreetmap.org/t/forststrassen-mit-schranken-korrektes-taggen/122347) and asked for their Ideas.
It seems that due to unclear laws they only tag the beginning of a forestal `highway` with `motor_vehicle=private` or `motor_vehicle=forestry`. This then often leads to an island in the middle that contains no access restrictions.

If the router then doesn't allow to pass trough a private `highway` you can't get in, nor out.
I don't see a way to integrate such a thing in a router without creating a mess, with 10 million special cases...

By playing around with differend OSM based apps i found that also OsmAnd solves this problem with a simple `allow private streets` switch as proposed [above](https://github.com/organicmaps/organicmaps/issues/9752#issuecomment-2506712099) as a solution.
